From dc96898057eacbf42aba009bd9ac7e0f8c7a00d8 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 28 Mar 2021 10:53:18 +0200
Subject: [PATCH 1/1] linker_lists: document ll_entry_ref parameters

Avoid 'make htmldocs' build warnings:

    ./include/linker_lists.h:224: warning:
    Function parameter or member '_type' not described in 'll_entry_ref'
    ./include/linker_lists.h:224: warning:
    Function parameter or member '_name' not described in 'll_entry_ref'
    ./include/linker_lists.h:224: warning:
    Function parameter or member '_list' not described in 'll_entry_ref'

Fixes: 851144350b6f ("linker_lists: Allow use in data structures")
Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 include/linker_lists.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/linker_lists.h b/include/linker_lists.h
index 81a280a884..2fea54c834 100644
--- a/include/linker_lists.h
+++ b/include/linker_lists.h
@@ -219,6 +219,10 @@
  *
  * This is like ll_entry_get(), but without the extra code, so it is suitable
  * for putting into data structures.
+ *
+ * @_type: C type of the list entry, e.g. 'struct foo'
+ * @_name: name of the entry
+ * @_list: name of the list
  */
 #define ll_entry_ref(_type, _name, _list)				\
 	((_type *)&_u_boot_list_2_##_list##_2_##_name)
-- 
2.30.2

